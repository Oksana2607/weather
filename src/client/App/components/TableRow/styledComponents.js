import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Row = styled.tr`
    &:hover {
        background-color: rgb(60, 142, 210);       
        font-weight: bold;        
    }    
    cursor: pointer;
    height: 2em;        
`;

export const RowCity = styled.td`
    padding: 1em;    
`;

export const StyledLink = styled(Link)`  
    text-decoration: none;   
    color: #000000;    
`;
