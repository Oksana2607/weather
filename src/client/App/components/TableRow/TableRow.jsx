import React from 'react';
import { Row, RowCity, StyledLink } from './styledComponents';

export const TableRow = (props) => {
  const { item } = props;

  return (
    <Row>
      <RowCity>
        <StyledLink to={`/${item.city}`}>
          {item.city}
        </StyledLink>
      </RowCity>
    </Row>
  );
};
