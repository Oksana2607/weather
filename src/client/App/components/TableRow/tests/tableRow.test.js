import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { TableRow } from '../TableRow';

configure({ adapter: new Adapter() });

describe('TableRow snapshot', () => {
  it('should render correctly', () => {
    const mockProps = {
      item: {}
    };

    const wrapper = shallow(
      <TableRow {...mockProps} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
