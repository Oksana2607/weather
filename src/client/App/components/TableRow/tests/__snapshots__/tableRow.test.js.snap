// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow snapshot should render correctly 1`] = `ShallowWrapper {}`;
